### PR TITLE
Skip LFS files in UberFilesizeTestEngine.

### PR DESCRIPTION
**Pull request by shatlykuber**
If "skip_lfs_files" config is tru for UberFilesizeTestEngine, then
it checks if the files exceeding the given filesize limit are tracked by
LFS. Large modified files that are tracked by LFS are ignored in this
unit test, as they will already get uploaded to LFS server.

### CODE REVIEW

File: src/unit/engine/UberFilesizeTestEngine.php [hunk 1]
```diff
@@ -69,6 +75,20 @@ private function checkFilesizes() {
         }
 
         if (count($large_files) > 0) {
+            if ($skip_lfs_files) {
+                // execute "git lfs" only if there are files over the limit,
+                // because lfs file list might be very big.
+                $lfs_files = $this->getLFSFiles();
+                $non_lfs_large_files = [];
```
Comment (robbertvanginkel) [inline src/unit/engine/UberFilesizeTestEngine.php]: You can rewrite this with [array_diff](http://php.net/manual/en/function.array-diff.php) into a simple set difference operation.

File: src/unit/engine/UberFilesizeTestEngine.php [hunk 2]
```diff
@@ -101,6 +121,23 @@ private function isExcluded($file, $exclude_paths) {
...
+        $return_code = 0;
+
+        exec(sprintf('git lfs ls-files'), $output, $return_code);
+
+        $lfs_files = [];
+        for ($i = 0; $i < count($output); $i++) {
+            // git lfs ls-files format is: <sha> * <filepath>
+            // for example:
+            // sha12345 * some/path/to.zip
+            array_push($lfs_files, explode(' ', $output[$i])[2]);
```
Comment (robbertvanginkel) [inline src/unit/engine/UberFilesizeTestEngine.php]: What if there's a space in the file path?

File: src/unit/engine/UberFilesizeTestEngine.php [hunk 3]
```diff
@@ -101,6 +121,23 @@ private function isExcluded($file, $exclude_paths) {
         return false;
     }
 
+    private function getLFSFiles() {
+        $output = [];
+        $return_code = 0;
+
+        exec(sprintf('git lfs ls-files'), $output, $return_code);
```
Comment (nguyentruongtho) [inline src/unit/engine/UberFilesizeTestEngine.php]: can just remove `sprintf`

Comment (nguyentruongtho) [inline src/unit/engine/UberFilesizeTestEngine.php]: optional: check $return_code as well?

File: src/unit/engine/UberFilesizeTestEngine.php (hunk 1 above)
Comment (shatlykuber) [inline src/unit/engine/UberFilesizeTestEngine.php]: done

File: src/unit/engine/UberFilesizeTestEngine.php (hunk 2 above)
Comment (shatlykuber) [inline src/unit/engine/UberFilesizeTestEngine.php]: done

File: src/unit/engine/UberFilesizeTestEngine.php (hunk 3 above)
Comment (shatlykuber) [inline src/unit/engine/UberFilesizeTestEngine.php]: done sprintf typo, exec should fail for nonzero return_code.

**Pull request merged**